exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +75,3066 @@ func TestStart(t *testing.T) {
 	testWg.Wait()
 }
 
-func TestGetSymbols(t *testing.T) {
-	t.Parallel()
-	_, err := g.GetSymbols(context.Background())
-	if err != nil {
-		t.Errorf("Gateio TestGetSymbols: %s", err)
-	}
+// Any tests below this line have the ability to impact your orders on the exchange. Enable canManipulateRealOrders to run them
+// ----------------------------------------------------------------------------------------------------------------------------
+func areTestAPIKeysSet() bool {
+	return g.ValidateAPICredentials(g.GetDefaultCredentials()) == nil
 }
 
-func TestGetMarketInfo(t *testing.T) {
+func TestCancelAllExchangeOrders(t *testing.T) {
 	t.Parallel()
-	_, err := g.GetMarketInfo(context.Background())
-	if err != nil {
-		t.Errorf("Gateio GetMarketInfo: %s", err)
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
 	}
-}
+	var orderCancellation = &order.Cancel{
+		OrderID:       "1",
+		WalletAddress: core.BitcoinDonationAddress,
+		AccountID:     "1",
+		Pair:          optionsTradablePair,
+		AssetType:     asset.Options,
+	}
+	resp, err := g.CancelAllOrders(context.Background(), orderCancellation)
 
-func TestSpotNewOrder(t *testing.T) {
-	t.Parallel()
+	if !areTestAPIKeysSet() && err == nil {
+		t.Error("Expecting an error when no keys are set")
+	}
+	if areTestAPIKeysSet() && err != nil {
+		t.Errorf("Could not cancel orders: %v", err)
+	}
 
-	if !areTestAPIKeysSet() || !canManipulateRealOrders {
-		t.Skip()
+	if len(resp.Status) > 0 {
+		t.Errorf("%v orders failed to cancel", len(resp.Status))
 	}
+}
 
-	_, err := g.SpotNewOrder(context.Background(),
-		SpotNewOrderRequestParams{
-			Symbol: "btc_usdt",
-			Amount: -1,
-			Price:  100000,
-			Type:   order.Sell.Lower(),
-		})
+func TestGetAccountInfo(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	_, err := g.UpdateAccountInfo(context.Background(), asset.Spot)
 	if err != nil {
-		t.Errorf("Gateio SpotNewOrder: %s", err)
+		t.Error("GetAccountInfo() error", err)
+	}
+	if _, err := g.UpdateAccountInfo(context.Background(), asset.Options); err != nil && !strings.Contains(err.Error(), "USER_NOT_FOUND") {
```

Comment:
Much like other reviews, if there is an API key check/realData check, don't do these `string.Contains` and just let the test fail